### PR TITLE
fix: can't connect to a cluster with private certificate

### LINKED ISSUE #547: Given an openshift cluster with private certificate and restricted rights I get told that it cannot list the namespaces
[kind/bug]

This was reported in https://github.com/redhat-developer/intellij-kubernetes/issues/600#issuecomment-1557066533

**Steps:**
1. ASSERT: have a cluster with a private certificate
2. EXEC: try to list the projects (expand the root node)

**Result:**
image
> The toolbox window shows the cluster, but the only item in the dropdown is "Unable to get namespaces: An error has occurred.", which is correct, because my account doesn't have the appropriate rights for oc get namespaces. It does for oc get projects though.

### PR DESCRIPTION
**Pull request by sbouchet**
## What is the purpose of this change? What does it change?
Openshift view can't list / createcompoennts on clusters using private certificates.

## Was the change discussed in an issue?
fixes #547

## How to test changes?
follow https://github.com/redhat-developer/intellij-kubernetes/issues/600#issuecomment-1562742941 to set up a cluster with private certificate ( CRC ), then please also modify the kubeconfig file to something like
```
- cluster:
    insecure-skip-tls-verify: true
    server: "https://api.crc.testing:6443"
  name: "api-crc-testing:6443"
```
without this manual change, odo can't access the cluster.

### CODE REVIEW

File: src/main/java/org/jboss/tools/intellij/openshift/utils/odo/OdoCli.java [hunk 1]
```diff
@@ -386,19 +392,11 @@ private void ensureDefaultOdoConfigFileExists() {
...
         String group = service.getApiVersion().substring(0, service.getApiVersion().indexOf('/'));
-        return new CustomResourceDefinitionContext.Builder()
-                .withName(service.getKind().toLowerCase() + "s." + group)
-                .withGroup(group)
-                .withScope(Scope.NAMESPACED.value())
-                .withKind(service.getKind())
-                .withPlural(service.getKind().toLowerCase() + "s")
-                .withVersion(version)
-                .build();
+        return new CustomResourceDefinitionContext.Builder().withName(service.getKind().toLowerCase() + "s." + group).withGroup(group).withScope(Scope.NAMESPACED.value()).withKind(service.getKind()).withPlural(service.getKind().toLowerCase() + "s").withVersion(version).build();
```
Comment (adietish) [inline src/main/java/org/jboss/tools/intellij/openshift/utils/odo/OdoCli.java:395]:
```java
.withPlural(service.getKind().toLowerCase() + "s"
```

fabric8 kubernetes-client has a nice utility to turn singular kinds to the plural form:
https://github.com/fabric8io/kubernetes-client/blob/main/kubernetes-model-generator/kubernetes-model-core/src/main/java/io/fabric8/kubernetes/api/Pluralize.java

**Review by adietish — APPROVED** [review]
other my suggestion for an improvement +1, LGTM

File: src/main/java/org/jboss/tools/intellij/openshift/utils/odo/OdoCli.java [hunk 2]
```diff
@@ -26,21 +26,26 @@
 import com.intellij.util.messages.MessageBusConnection;
 import com.redhat.devtools.intellij.common.kubernetes.ClusterHelper;
 import com.redhat.devtools.intellij.common.kubernetes.ClusterInfo;
+import com.redhat.devtools.intellij.common.utils.ConfigHelper;
 import com.redhat.devtools.intellij.common.utils.ExecHelper;
 import com.redhat.devtools.intellij.common.utils.NetworkUtils;
+import com.redhat.devtools.intellij.kubernetes.model.client.ssl.IDEATrustManager;
```
Comment (adietish) [inline src/main/java/org/jboss/tools/intellij/openshift/utils/odo/OdoCli.java:32]: Should we move this to common?

Comment (sbouchet) [reply src/main/java/org/jboss/tools/intellij/openshift/utils/odo/OdoCli.java:32]: Agreed, as it's common to kubernetes and openshift.

Comment (sbouchet) [reply src/main/java/org/jboss/tools/intellij/openshift/utils/odo/OdoCli.java:32]: created https://github.com/redhat-developer/intellij-common/issues/200

**Review by adietish — APPROVED** [review]
tested it with the latest cluster set up by our group in "forum-ui-clusters" at https://api.akundu-XXXXXX.devcluster.openshift.com:6443/. In cmd-line this cluster can only be connected to by using the `--insecure-skip-tls-verify` flag (which then adds the above property to the kubeconfig). Could connect to it using intellij-openshift-connector and install helm chart to it. 
LGTM

**Pull request merged**